### DISCUSSION TITLE
provide a user friendly config_entry reload  service

### DISCUSSION THREAD

**Discussion by vingerha** [frenck/spook#513] [Ideas]
Core provides a service "homeassistant.reload_config_entry" that requires a target_id and a config_entry_id
The target is not (always) needed but is required and the config_entry_id needs digging into the core.config_entries
It would be nice that one could select te config entry from a list and then fire the reload....

EDIT: explaining a bit more my challenge.
I have multiple config_entries that each have multiple entities. Based on the location change of the device_tracker, connected to the config_entry, I need to update all entities. As I cannot identify this change from entity>entry>device I would like to trigger this from a device location change, then call the config_entry reload which then updates the entities.
I was thinking about writing my own service call but that would not serve a larger audience

**Comment by frenck** [comment]
> The target is not (always) needed 

Uh... the service targets something, it is a mandatory key. I'm not following this part of your story.

> config_entry_id needs digging into the core.config_entries

Not really, you can get it from the UI. It is documented as well.
As a matter of fact, you should never go into `core.config_entries` at all.

![CleanShot 2023-12-31 at 16 43 14@2x](https://github.com/frenck/spook/assets/195327/b11b6ea0-ff9b-4396-b757-9c66d812618e)

Alternatively, you can go to the UI, view the device page, and the ID will be in the URL of your browser.

Or, last but not least, the UI provides selectors for these everywhere.


TL;DR: There are many methods available to get it. From the UI, it will all provide an dropdown to select it from.

From the Spook end of things, there is no influence possible in this part.

../Frenck

**Comment by vingerha** [comment]
I have multiple entities that refer to the same config_entry and multiple config_entries to one device_tracker. These entities also disappear (as planned) so I cannot (!) use the entity/device as starter which makes this service impossible as-it-is. I noticed that I can use 'any' entity and supply the config_entry_id but finding the config_entry_id from .storage is a no-go for me...
Anyhow, I found a solution via config_entries.async_entries(DOMAIN) and then iterating

**Comment by vingerha** [comment]
And I feel I still haven't explained the use case in detail.
It is for my gtfs2 integration, I am collecting stop around 'me' (person or zone) and any upcoming departures per stop
Changing elements are time (for departure) and location (for stops).
It collects the data based on the location lat/lon and each stop gets its own entity plus the upcoming departures as attributes. When a person moves to the other end of town, the previous stops are no longer of importance and need to be removed, and new ones created based on the new location. Easiest is to reload the config_entry based on a trigger or action....so I want to use the existing service with fixed data and the only fixed element here is either the person or the zone as the entities change.
